internal/providers/terraform/aws/s3_bucket_lifecycle_configuration.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"github.com/infracost/infracost/internal/resources/aws"
+
+	"github.com/infracost/infracost/internal/schema"
+)
+
+func getS3BucketLifecycleConfigurationRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_s3_bucket_lifecycle_configuration",
+		RFunc: newS3BucketLifecycleConfigurationResource,
+	}
+}
+
+func newS3BucketLifecycleConfigurationResource(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	storageClassNames := map[string]string{
```

Comment:
I kept standard class on the S3 bucket: I couldn't find a way to detect the provider version, and standard class is added on the resource by default, I couldn't add any conditions on that. It looks weird in the output though, so maybe you have any better ideas? When we decide to stop supporting v3, we'll need to remove all that code from S3 bucket resource, along with moving that standard class here.

Reply:
Should we have it on both? An S3 bucket doesn't need a lifecycle configuration so the costs might be on either.

Reply:
Since they're all usage-based costs would it not just show the prices unless you add usage data? And so it would only show costs against the resource you added the usage data to.

Reply:
Yep, it's up to the user to decide where to add the usage params. Sounds good 👍

Reply:
Fixed in https://github.com/infracost/infracost/pull/1370/commits/addefa9c6486951666d492d08fb03c4de96d1a05. Will fixup after code review.